community/knative/charts/serving/charts/monitoring/charts/elasticsearch/templates/fluentd-pod.yaml

```diff
@@ -0,0 +1,72 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels: {{ include "labels" . | nindent 4 }}
+    addonmanager.kubernetes.io/mode: Reconcile
+    app: {{ .Values.fluentd.name | quote }}
+    kubernetes.io/cluster-service: "true"
+    version: {{ .Values.fluentd.image.tag | quote }}
+  name: {{ .Values.fluentd.name | quote }}
+  namespace: "knative-monitoring"
+spec:
+  selector:
+    matchLabels:
+      app: {{ .Values.fluentd.name | quote }}
+      version: {{ .Values.fluentd.image.tag | quote }}
+  template:
+    metadata:
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ""
+      labels: {{ include "labels" . | nindent 8 }}
+        app: {{ .Values.fluentd.name | quote }}
+        kubernetes.io/cluster-service: "true"
+        version: {{ .Values.fluentd.image.tag | quote }}
+    spec:
+      containers:
+      - env:
+        - name: FLUENTD_ARGS
+          value: --no-supervisor -q
+        image: "{{ .Values.fluentd.image.repo }}/{{ .Values.fluentd.image.name }}:{{ .Values.fluentd.image.tag }}"
+        name: {{ .Values.fluentd.name | quote }}
+        resources:
+          limits:
+            memory: 500Mi
+          requests:
+            cpu: 100m
+            memory: 200Mi
+        volumeMounts:
+        - mountPath: /var/log/containers
+          name: varlogcontainers
+          readOnly: true
+        - mountPath: /var/log/pods
+          name: varlogpods
+          readOnly: true
+        - mountPath: /var/lib/docker/containers
+          name: varlibdockercontainers
+          readOnly: true
+        - mountPath: /host/lib
+          name: libsystemddir
+          readOnly: true
+        - mountPath: /etc/fluent/config.d
+          name: config-volume
+      nodeSelector:
+        beta.kubernetes.io/{{ .Values.fluentd.name }}-ready: "true"
+      priorityClassName: system-node-critical
+      serviceAccountName: {{ .Values.fluentd.serviceAccount.name | quote }}
+      terminationGracePeriodSeconds: 30
+      volumes:
+      - hostPath:
```

Comment:
This is kept "as is" to keep this chart inline with what the Knative community delivers. This will  make maintaining the chart easier as new release are rolled out.

However, we will look into this further and try to contribute these best practices in the Knative community and then bring in those changes over to this chart in time.